jgrapht-core/src/test/java/org/jgrapht/util/DoublyLinkedListTest.java

```diff
@@ -0,0 +1,1740 @@
+/*
+ * (C) Copyright 2020-2020, by Hannes Wellmann and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+package org.jgrapht.util;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+
+import java.util.*;
+
+import org.jgrapht.util.DoublyLinkedList.*;
+import org.junit.*;
+import org.junit.rules.*;
+import org.junit.runner.*;
+import org.junit.runners.*;
+import org.junit.runners.Parameterized.*;
+
+/**
+ * @author Hannes Wellmann
+ *
+ */
+@RunWith(Parameterized.class)
+public class DoublyLinkedListTest
+{
+    private static final int MAX_LIST_SIZE = 8;
+
+    @Parameters(name = "List with size {0}")
+    public static Object[] getListSizes()
+    {
+        Object[][] parameterSets = new Object[MAX_LIST_SIZE + 1][];
+        for (int size = 0; size < MAX_LIST_SIZE + 1; size++) {
+
+            List<Object> elements = new ArrayList<>(size);
+            for (int i = 0; i < size; i++) {
+                elements.add("obj" + i);
+            }
+            if (size >= 6) { // make two elements equal
+                elements.set(3, new String("obj2")); // create equal new String-object
```

Comment:
`new String(...)` is redundant

Reply:
I used `new String()` intentionally to enforce another `String` object, because the compiler will likely use the same `String`-object for each occurrence of the literal "obj2".
In the tests I use identity checks instead of equals() to check if the expected is a particular object. So it can be tested precisely if the List behaves as expected regarding equal and same objects.